packages/plugin-npm-cli/sources/commands/npm/publish.ts

```diff
@@ -99,28 +117,42 @@ export default class NpmPublishCommand extends BaseCommand {
       await packUtils.prepareForPack(workspace, {report}, async () => {
         const files = await packUtils.genPackList(workspace);
 
-        for (const file of files)
-          report.reportInfo(null, file);
+        for (const file of files) {
+          report.reportInfo(null, npath.fromPortablePath(file));
+          report.reportJson({file: npath.fromPortablePath(file)});
+        }
 
         const pack = await packUtils.genPackStream(workspace, files);
         const buffer = await miscUtils.bufferStream(pack);
-
         const gitHead = await npmPublishUtils.getGitHead(workspace.cwd);
 
+        // Determine provenance
         let provenance = false;
-        if (workspace.manifest.publishConfig && `provenance` in workspace.manifest.publishConfig) {
+        if (workspace.manifest.publishConfig && `provenance` in workspace.manifest.publishConfig)
           provenance = Boolean(workspace.manifest.publishConfig.provenance);
-          if (provenance) {
-            report.reportInfo(null, `Generating provenance statement because \`publishConfig.provenance\` field is set.`);
-          } else {
-            report.reportInfo(null, `Skipping provenance statement because \`publishConfig.provenance\` field is set to false.`);
-          }
-        } else if (this.provenance) {
+        else if (this.provenance)
           provenance = true;
-          report.reportInfo(null, `Generating provenance statement because \`--provenance\` flag is set.`);
-        } else if (configuration.get(`npmPublishProvenance`)) {
+        else if (configuration.get(`npmPublishProvenance`))
           provenance = true;
-          report.reportInfo(null, `Generating provenance statement because \`npmPublishProvenance\` setting is set.`);
+
+        // Report provenance decision
+        let message = ``;
+        if (workspace.manifest.publishConfig && `provenance` in workspace.manifest.publishConfig)
+          message = provenance
+            ? `Generating provenance statement because \`publishConfig.provenance\` field is set.`
+            : `Skipping provenance statement because \`publishConfig.provenance\` field is set to false.`;
+        else if (this.provenance)
+          message = `Generating provenance statement because \`--provenance\` flag is set.`;
+        else if (provenance)
+          message = `Generating provenance statement because \`npmPublishProvenance\` setting is set.`;
```

Comment:
I think the assignment of `message` should be moved back into the same conditional branches as the one that sets `provenance`, otherwise there's a high chance those branches will end up out of sync.